snap/snapcraft.yaml

```diff
@@ -0,0 +1,202 @@
+  icon-themes:
+    interface: content
+    target: $SNAP/data-dir/icons
+    default-provider: gtk-common-themes:icon-themes
+  sound-themes:
+    interface: content
+    target: $SNAP/data-dir/sounds
+    default-provider: gtk-common-themes:sounds-themes
+
+parts:
+    musescore:
+        source: https://github.com/musescore/MuseScore
+        source-type: git
+        plugin: make
+        override-build: |
+            make revision
+            QT_SELECT=qt5 snapcraftctl build
+        override-pull: |
+            snapcraftctl pull
+            "$SNAPCRAFT_STAGE"/scriptlets/selective-checkout \
+                        --release-tag-prefix='v' \
+                        --stable-tag-pattern=[[:digit:]]+$ \
+                        --force-stable
+        build-packages:
+            #- extra-cmake-modules
+            #- doxygen
+            # From https://musescore.org/en/handbook/developers-handbook/compilation/compile-instructions-ubuntu-1604lts-and-1804lts-git
+            # Tools
+            - cmake
+            - g++
+            # Audio & others
+            - libasound2-dev
+            - portaudio19-dev
+            - libmp3lame-dev
+            - libsndfile1-dev
+            - libportmidi-dev
+            - libssl-dev
+            - libpulse-dev
+            - libfreetype6-dev
+            - libfreetype6
+            # «Failed to find "GL" in "".»
+            - libdrm-dev
+            - libgl1-mesa-dev
+            - libegl1-mesa-dev
+            # QT
+            - qtbase5-dev
+            - qttools5-dev
+            - qttools5-dev-tools
+            - qtwebengine5-dev
+            - qtscript5-dev
+            - libqt5xmlpatterns5-dev
+            - libqt5svg5-dev
+            - libqt5webkit5-dev
+            - qtquickcontrols2-5-dev
+        make-parameters:
+            - PREFIX=/usr
+            - UPDATE_CACHE=FALSE
+            # TODO: This shouldn't be necessary
+            # https://musescore.org/en/comment/884393
+            - BUILD_WEBENGINE=off
+        stage-packages:
+            - libqt5core5a
+            - libqt5gui5
+            - libqt5network5
+            - libqt5xml5
+            - libqt5xmlpatterns5
+            - libqt5svg5
+            - libqt5printsupport5
+            - libqt5webkit5
+            - libcanberra-gtk-module
+            - overlay-scrollbar-gtk2
+            - unity-gtk2-module
+            - libatk-adaptor
+            - libgail-common
+            - libmp3lame0
+            - libportmidi0
+            - libvorbis0a
+            - libsndfile1
+            - libportaudio2
+            - libpulse0
+            - libjack0
+            - libasyncns0
+            - libflac8
+            - qtwayland5
+            - libqt5help5
+            - libqt5sql5
+            # Palettes need these, see:
+            # https://forum.snapcraft.io/t/missing-dependency-for-musescore/14884
+            - qml-module-qtgraphicaleffects
+            - qml-module-qtquick-controls2
```

Comment:
I believe "qml-module-qtquick-controls" is neede too, see e.g. https://musescore.org/en/node/301501#comment-980984

Reply:
I've just added it! Thanks!

Reply:
The version with this installed is on the beta/edge channel of the snap, but now this error appears when trying to load the `abc_import.qml` plugin: `QQmlComponent: Component is not ready`
Any ideas?

Reply:
OK, the plugin loader showed this error: `Line 21: module "QtQuick.Dialogs" is not installed` when trying to execute the plugin, so at least package `qml-module-qtquick-dialogs` is also needed.

Reply:
No I get these errors:
```
   line 47: module "Qt.labs.folderlistmodel" is not installed
   line 48: module "Qt.labs.settings" is not installed
   line 47: module "Qt.labs.folderlistmodel" is not installed
   line 48: module "Qt.labs.settings" is not installed
```
I guess we need more packages...

Reply:
Those 2 are surely  needed indeed

Reply:
OK, this is now working and on the snap stable channel. Thanks for the tips @Jojo-Schmitz !